src/auth/Crypto.cc

```diff
@@ -35,30 +36,58 @@
 #include "common/debug.h"
 #include <errno.h>
 
-int get_random_bytes(char *buf, int len)
+// use a mersenne_twister_engine for random numbers. this engine is faster than
+// std::random_device, and generates higher quality randomness than the other
+// alternatives
+using base_random_engine = std::mt19937;
+
+// wrap with an independent_bits_engine to allow get_random_bytes() to sample
+// a single byte at a time. uniform_int_distribution can't be used for this,
+// because it is undefined for integer types smaller than short
+using streaming_random_engine = std::independent_bits_engine<
+    base_random_engine, 8, unsigned char>;
+
+// engine factory function
+static streaming_random_engine make_seeded_random_engine()
```

Comment:
Alternatively, you could consider adding SFINAE selection in ceph::util::random() to set up this engine. See randomize_rng()-- it may be trickier to shoehorn in than necessary, but OTOH it would be nice to be able to go through it here. OTOH, that library isn't really mean to be suited to specialized tasks, which this may be.

Reply:
Oh! std::random_device()'s constructor throws if it can't initialize. (At least in the case of libc++, that means on a failure to open /dev/random.)